### PR TITLE
Gets AttackMove to work with 'Defend' stance

### LINKED ISSUE #8184: AttackMove is broken for some unit types with "Defend" stance
While working on https://github.com/OpenRA/OpenRA/compare/bleed...obrakmann:default-stance-defend I noticed that AttackMove is broken for E3s when their stance is set to "Defend" (in ra at least, didn't test the other mods or other units yet). They acquire a target, but aren't close enough to fire and won't move in range. E1s don't seem to have that problem.

screenshot from 2015-05-17 20 34 06
screenshot from 2015-05-17 20 31 43

I can reproduce this in release-20141029, so it's not really a recent bug.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/15506690-attackmove-is-broken-for-some-unit-types-with-defend-stance?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by matija-hustic**
Fixes #8184.

### CODE REVIEW

**Comment by obrakmann** [comment]
This does indeed fix it, well done! I need to play some more to see if there are some unwanted side effects, but I'm already looking forward to dusting off my defend-as-default-stance branch :)

**Comment by obrakmann** [comment]
Test games went fine. :+1:

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 1]
```diff
@@ -170,7 +171,9 @@ Actor ChooseTarget(Actor self, WDist range, bool allowMove)
 				// criterion
 				.Select(a => new KeyValuePair<Armament, Actor>(
 					attack.ChooseArmamentsForTarget(Target.FromActor(a), false)
-						.Where(arm => allowMove || arm.MaxRange().Length > (a.CenterPosition - self.CenterPosition).Length)
+						.Where(arm => allowMove
+							|| (Target.FromActor(a).IsInRange(self.CenterPosition, arm.MaxRange())
+							&& !Target.FromActor(a).IsInRange(self.CenterPosition, arm.Weapon.MinRange)))
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: Can you pull out a target variable instead of having three `Target.FromActor(a)`?

**Comment by matija-hustic** [comment]
Updated

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/38d529f50cacc39fc61eec21797ad7f5e8d1fccc...18bc7b278cb1999ca91f8d7293d53844bdfc003a)

**Pull request merged**